pybind_interface/pybind_main.cpp

```diff
@@ -448,297 +448,199 @@ std::vector<std::complex<float>> qtrajectory_simulate(const py::dict &options) {
   return amplitudes;
 }
 
-// Simulate from a "pure" starting state.
-qsim::Simulator<For>::State _noiseless_sim_from_input_state(
-    const py::dict &options, uint64_t input_state) {
+// Helper class for simulating circuits of all types.
+struct SimulatorHelper {
```

Comment:
I would prefer stateless design. But it's OK as it is. Make the content of this struct private except for `simulate_fullstate` and `simulate_expectation_values`.

Reply:
Done. Since private variables cannot be brace-initialized, `SimulatorHelper::from_options` has been converted into an explicitly-defined (but private) constructor, with the default constructor deleted.